tests/integration/security/rbac/util/rbac_util.go

```diff
@@ -39,58 +40,56 @@ type TestCase struct {
 // *** For TCP: EOF error
 // * If the policy is allow:
 // *** Response code is 200
-func CheckRBACRequest(tc TestCase) error {
+func (tc TestCase) Check() error {
 	req := tc.Request
-	ep := req.To.EndpointForPort(req.Port)
-	if ep == nil {
-		return fmt.Errorf("cannot get upstream endpoint for connection test %v", req)
-	}
 
 	headers := make(http.Header)
 	if len(tc.Jwt) > 0 {
 		headers.Add("Authorization", "Bearer "+tc.Jwt)
 	}
+	tc.Request.Options.Headers = headers
 
-	resp, err := req.From.Call(ep, apps.AppCallOptions{Protocol: req.Protocol, Path: req.Path, Headers: headers})
-	if err != nil && !strings.Contains(err.Error(), "EOF") {
-		return fmt.Errorf("connection error with %v", err)
-	}
+	resp, err := req.From.Call(tc.Request.Options)
 
 	if tc.ExpectAllowed {
-		if !(len(resp) > 0 && resp[0].Code == connection.AllowHTTPRespCode) {
-			return fmt.Errorf("%s to %s:%d%s using %s: expected allow, actually deny",
-				req.From.Name(), req.To.Name(), req.Port, req.Path, req.Protocol)
+		if err == nil {
+			err = resp.CheckOK()
+		}
+		if err != nil {
+			return fmt.Errorf("%s to %s:%s%s using %s: expected allow but was denied: %v",
+				req.From.Config().Service,
+				req.Options.Target.Config().Service,
+				req.Options.PortName,
+				req.Options.Path,
+				req.Options.Scheme,
+				err)
 		}
 	} else {
-		if req.Port == connection.TCPPort {
-			if !strings.Contains(err.Error(), "EOF") {
-				return fmt.Errorf("%s to %s:%d%s using %s: expected deny with EOF error, actually %v",
-					req.From.Name(), req.To.Name(), req.Port, req.Path, req.Protocol, err)
+		if err != nil && !strings.Contains(err.Error(), "EOF") {
```

Comment:
This seems not right, we only need to check if the err is "EOF" for TCP connection.

Reply:
I don't believe that I changed the original logic, but it's possible.